src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,46 @@
 package core.mate.academy.service;
 
-/**
- * Your implementation of MachineService.
- */
-public class MachineServiceImpl {
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Track;
+import java.util.ArrayList;
+import java.util.List;
+
+public class MachineServiceImpl implements MachineService<Machine> {
+    @Override
+    public List<Machine> getAll(Class<? extends Machine> type) {
+        if (type.equals(Excavator.class)) {
+            List<Excavator> excavatorList = new ExcavatorProducer().get();
+            return new ArrayList<>(excavatorList);
+        }
+        if (type.equals(Bulldozer.class)) {
+            List<Bulldozer> bulldozerList = new BulldozerProducer().get();
+            return new ArrayList<>(bulldozerList);
+        }
+        if (type.equals(Track.class)) {
+            List<Track> trackList = new TrackProducer().get();
+            return new ArrayList<>(trackList);
+        }
+        return null;
```

Comment:
it's bad practice to return null, may produce NPE when using code, let's throw exception here or return empty list